src/librustc_allocator/expand.rs

```diff
@@ -142,23 +144,18 @@ impl<'a> Folder for ExpandAllocatorDirectives<'a> {
         let module = f.cx.monotonic_expander().fold_item(module).pop().unwrap();
 
         // Return the item and new submodule
-        smallvec![item, module]
+        Some(vec![item.clone(), module])
     }
 
     // If we enter a submodule, take note.
-    fn fold_mod(&mut self, m: Mod) -> Mod {
+    fn visit_mod(&mut self, m: &mut Mod, _s: Span, _attrs: &[Attribute], _n: NodeId) {
         debug!("enter submodule");
         self.in_submod += 1;
-        let ret = fold::noop_fold_mod(m, self);
+        let ret = visit_mut::walk_mod(self, m);
         self.in_submod -= 1;
         debug!("exit submodule");
         ret
     }
-
-    // `fold_mac` is disabled by default. Enable it here.
-    fn fold_mac(&mut self, mac: Mac) -> Mac {
-        fold::noop_fold_mac(mac, self)
-    }
```

Comment:
To confirm, `MutVisitors` should still panic when it encounters a macro, but allocator expansion is done after all macros are already expanded, so this is ok?

Reply:
I don't think any macros are around by this point. I think this function was just copied and pasted from another `Folder` impl.